src/EventListener/SerializerViewListener.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\EventListener;
+
+use ApiPlatform\Core\Api\RequestAttributesExtractor;
+use ApiPlatform\Core\Exception\RuntimeException;
+use ApiPlatform\Core\Serializer\SerializerContextBuilderInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Serializes data.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class SerializerViewListener
+{
+    private $serializer;
+    private $serializerContextBuilder;
+
+    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder)
+    {
+        $this->serializer = $serializer;
+        $this->serializerContextBuilder = $serializerContextBuilder;
+    }
+
+    /**
+     * Serializes the data to the requested format.
+     *
+     * @param GetResponseForControllerResultEvent $event
+     */
+    public function onKernelView(GetResponseForControllerResultEvent $event)
+    {
+        $controllerResult = $event->getControllerResult();
+        $request = $event->getRequest();
+
+        if ($controllerResult instanceof Response) {
+            return;
+        }
+
+        try {
+            $extractedAttributes = RequestAttributesExtractor::extractAttributes($request);
+        } catch (RuntimeException $e) {
+            return;
```

Comment:
When the request is not managed by API Platform (for instance when the attribute `_api_format` is not set).